src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,50 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.TaskList;
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.Ui;
+
+import java.io.File;
+
+
+public class Duke {
+    private final Parser PARSER;
+    private final Ui UI;
+
+    public static void main(String[] args) {
+        String filePath = "data" + File.separatorChar + "duke-storage.txt";
+        new Duke(filePath).run();
+    }
+
+    public Duke(String filePathToStorage) {
+        Storage DUKE_STORE = new Storage(filePathToStorage);
+        TaskList TASKLIST = TaskList.of(DUKE_STORE);
+        this.PARSER = Parser.initialize(TASKLIST);
+        this.UI = new Ui();
```

Comment:
Should `this.PARSER` and `this.UI` simply be `PARSER` and `UI` as they are not shadowed by a method or constructor parameters? I noticed this issue in several other places too, maybe it is something for you to consider?
```suggestion
        PARSER = Parser.initialize(TASKLIST);
        UI = new Ui();
```

Reply:
haha yea I was thinking about changing it, cos in cs2030 they said put this.x for like everything so kinda stuck to it, will change it tho thx